server/java/src/main/java/com/godaddy/asherah/grpc/ServerApp.java

```diff
@@ -0,0 +1,110 @@
+package com.godaddy.asherah.grpc;
+
+import com.godaddy.asherah.appencryption.SessionFactory;
+import com.godaddy.asherah.appencryption.kms.KeyManagementService;
+import com.godaddy.asherah.appencryption.persistence.Metastore;
+import com.godaddy.asherah.crypto.CryptoPolicy;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+import picocli.CommandLine;
+
+class ServerApp implements Callable<Void> {
+  static class MetastoreTypes extends ArrayList<String> {
+    MetastoreTypes() {
+      super(Arrays.asList(Constants.METASTORE_INMEMORY, Constants.METASTORE_JDBC, Constants.METASTORE_DYNAMODB));
+    }
+  }
+
+  static class KmsTypes extends ArrayList<String> {
+    KmsTypes() {
+      super(Arrays.asList(Constants.KMS_STATIC, Constants.KMS_AWS));
+    }
+  }
+
+  // Options to configure the metastore
+  @CommandLine.Option(names = "--metastore-type", defaultValue = "${env:ASHERAH_METASTORE_MODE}",
```

Comment:
I couldn't find a clean way of handling both application defaults and environment variables. However, this will be revisited with https://github.com/godaddy/asherah/issues/143.